sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -226,7 +226,14 @@ public static boolean isEnableMultipleWriteLocations(DatabaseAccount account) {
 
     public static <E extends CosmosClientException> void setRequestHeaders(CosmosClientException cosmosClientException,
                                                                            Map<String, String> requestHeaders) {
-        cosmosClientException.requestHeaders = requestHeaders;
+        Map<String, String> requestHeadersMap = new HashMap<>();
```

Comment:
why not removing the auth header field directly from the requestHeaders rather than making a copy of it? For the auth entry we are suppose to re-populate that every time rather than reusing it so if there are failures because of it then we have other bugs to fix :-)